deploy/cdk/lib/static-host/pipeline-s3-sync.ts

```diff
@@ -12,15 +12,27 @@ export interface IPipelineS3SyncProps extends PipelineProjectProps {  /**
   readonly targetStack: string
 
   /**
-   * Artifact that contains the build which needs to be synced to the s3 bucket.
-   * Presumably the output from a previous codebuild project.
+   * Artifact that contains the files which need to be built by the code.
+   * Presumably the output from a github pull
    */
   readonly inputBuildArtifact: Artifact;
 
   /**
-   * Subdirectory of files to sync. Optional; will sync everything by default.
-   */
-  readonly subdirectory?: string
+  * The name of the index that should be created for the website in elasticsearch
+  */
+  readonly searchIndex?: string
+
+  /**
+  * the ssm path to the full elasticsearch domain that we want a search index on\
+  * example: https://search-jon-test-sites-xnwpt33aguihqeotpz7yp6zp5m.us-east-1.es.amazonaws.com
+  */
+  readonly esEndpointParamPath?: string
+
+  /**
+  * the domain name part that is used for permissions to the elastic search domain.
+  * example: jon-test-sites
+  */
+  readonly elasticSearchDomainName?: string
```

Comment:
Looking at the usage of these, they seem required. I'd remove the optionals to catch issues before deployment time